src/core/Akka/Actor/Message.cs

```diff
@@ -12,7 +12,7 @@ namespace Akka.Actor
     /// <summary>
     /// Envelope class, represents a message and the sender of the message.
     /// </summary>
-    public struct Envelope
+    public sealed class Envelope
```

Comment:
This is technically not a "binary compatible change" but I don't think it'll have any breaking changes on end-users:

1. `Envelope`, although a `public` type, is not accessed directly by users typically;
2. Even if it was, none of the public signatures have changed as this change is source-compatible.

Reply:
As a (I think I'm at) _tertiary_ curveball, did we see what happened if we made it a `readonly struct`? Will admit I am not exactly confident it will be better but I still wonder about certain locality aspects around dispatch.